docs/root/configuration/http/http_filters/local_rate_limit_filter.rst

```diff
@@ -103,6 +103,83 @@ The route specific configuration:
 Note that if this filter is configured as globally disabled and there are no virtual host or route level
 token buckets, no rate limiting will be applied.
 
+Using Descriptors to rate limit on
+----------------------------------
+
+Descriptors can be used to override local rate limiting based on presence of certain descriptors/route actions.
+A routes :ref:`rate limit action <envoy_v3_api_msg_config.route.v3.RateLimit>` is used to match up a
+:ref:`descriptor <envoy_v3_api_msg_extensions.common.ratelimit.v3.RateLimitDescriptor>` in the filter config descriptor list.
+The descriptor's rate limit override value is used to specify the rate limit units to use if this descriptor is a match
+otherwise token bucket default config is used.
+
+Example filter configuration using descriptors is as follows:
+
+.. code-block:: yaml
+
+  route_config:
+    name: local_route
+    virtual_hosts:
+    - name: local_service
+      domains: ["*"]
+      routes:
+      - match: { prefix: "/foo" }
+        route: { cluster: service_protected_by_rate_limit }
+        typed_per_filter_config:
+          envoy.filters.http.local_ratelimit:
+            "@type": type.googleapis.com/envoy.extensions.filters.http.local_ratelimit.v3.LocalRateLimit
+            stat_prefix: test
+            token_bucket:
+              max_tokens: 1000
+              tokens_per_fill: 1000
+              fill_interval: 60s
+            filter_enabled:
+              runtime_key: test_enabled
+              default_value:
+                numerator: 100
+                denominator: HUNDRED
+            filter_enforced:
+              runtime_key: test_enforced
+              default_value:
+                numerator: 100
+                denominator: HUNDRED
+            response_headers_to_add:
+              - append: false
+                header:
+                  key: x-test-rate-limit
+                  value: 'true'
+            descriptors:
+            - entries:
+              - key: client_id
+                value: foo
+              - key: path
+                value: /foo/bar
+              limit:
+                requests_per_unit: 10
+                unit: MINUTE
+            - entries:
+              - key: client_id
+                value: foo
+              - key: path
+                value: /foo/bar2
+              limit:
+                requests_per_unit: 100
```

Comment:
Could use the same syntax as token bucket above here? It feels inconsistent to have two ways to specify upper bound in the same config. The max tokens is also missing.

Reply:
It's reusing existing Descriptor object that http global rate limit uses and hence there is no max tokens and token bucket here...

Reply:
You can define a new message if you want, up to you. I agree with @kyessenov that it would be better to be consistent and the message is very simple so should be fine?

Reply:
okay.. so how about adding TokenBucket as a message in RateLimitDescriptor.
Global one can use overrride units and local token bucket.. I updated just the api proto.. ptal and I will update rest of the code using it.

Reply:
No, I don't think that makes sense. Please add a new LocalRateLimitDescriptor. That can include a RateLimitDescriptor and a token bucket config for example.

/wait

Reply:
Added... LocalRateLimitDescriptor with Entries and token bucket... don't think it makes sense to include whole RateLimitDescriptor and get RateLimitOverride which is specific to global?

Reply:
No wait, I would need whole RateLimitDescriptor in LocalRateLimitDescriptor, if I want to keep code common between global and local for populating these descriptors from RouteEntry

Reply:
updated api change..ptal